server/handle.go

```diff
@@ -151,14 +151,10 @@ func (l *listener4) HandleMsg4(buf []byte, oob *ipv4.ControlMessage, _peer net.A
 		} else if req.IsBroadcast() {
 			peer = &net.UDPAddr{IP: net.IPv4bcast, Port: dhcpv4.ClientPort}
 		} else {
-			// FIXME: we're supposed to unicast to a specific *L2* address, and an L3
-			// address that's not yet assigned.
-			// I don't know how to do that with this API...
-			//peer = &net.UDPAddr{IP: resp.YourIPAddr, Port: dhcpv4.ClientPort}
-			log.Warn("Cannot handle non-broadcast-capable unspecified peers in an RFC-compliant way. " +
-				"Response will be broadcast")
+			//sends a layer2 frame so that we can define the destination MAC address
+			sendEthernet(l.Interface, resp)
```

Comment:
`l.Interface` here isn't necessarily non-nil if the configuration doesn't restrict coredhcp to listen on a specific interface; you'll need to use the same logic as lines 161-174 to find the right interface for the return packet

Reply:
Should be OK now.